cluster-autoscaler/core/scale_down.go

```diff
@@ -103,6 +103,93 @@ func (n *NodeDeleteStatus) SetDeleteInProgress(status bool) {
 	n.deleteInProgress = status
 }
 
+type resourceName string
+type scaleDownResourcesLimits map[resourceName]int64
+type scaleDownResourcesDelta map[resourceName]int64
+
+func computeScaleDownResourcesLeftLimits(nodes []*apiv1.Node, cp cloudprovider.CloudProvider, timestamp time.Time) (scaleDownResourcesLimits, errors.AutoscalerError) {
+	resourceLimiter, err := cp.GetResourceLimiter()
+	if err != nil {
+		return scaleDownResourcesLimits{}, errors.ToAutoscalerError(
+			errors.CloudProviderError,
+			err)
+	}
+
+	totals := make(map[resourceName]int64)
+	totals[cloudprovider.ResourceNameCores], totals[cloudprovider.ResourceNameMemory] = calculateCoresAndMemoryTotal(nodes, timestamp)
+
+	for gpuName, gpuCount := range calculateGpusTotal(nodes, cp, timestamp) {
+		totals[resourceName(gpuName)] = gpuCount
+	}
+
+	resultScaleDownLimits := make(scaleDownResourcesLimits)
+	for _, resource := range resourceLimiter.GetResources() {
+		total := totals[resourceName(resource)]
+		min := resourceLimiter.GetMin(resource)
+
+		// we put only actual limits into final map. No entry means no limit.
+		if min > 0 {
+			resultScaleDownLimits[resourceName(resource)] = total - min
+		}
+	}
+	return resultScaleDownLimits, nil
+}
+
+func noLimitsOnResources() scaleDownResourcesLimits {
+	return nil
+}
+
+func copyScaleDownResourcesLimits(source scaleDownResourcesLimits) scaleDownResourcesLimits {
+	copy := scaleDownResourcesLimits{}
+	for k, v := range source {
+		copy[k] = v
+	}
+	return copy
+}
```

Comment:
Missing newline between functions (I'm really surprised gofmt didn't pick that up...)